src/diff.rs

```diff
@@ -3,12 +3,14 @@ pub async fn diff() -> fpm::Result<()> {
     let snapshots = fpm::snaphot::get_latest_snapshots(config.root.as_str())?;
 
     for doc in fpm::process_dir(config.root.as_str(), &config).await? {
-        if doc.id.starts_with(".history") {
-            continue;
-        }
-        if let Some(diff) = get_diffy(&doc, &snapshots).await? {
-            println!("diff: {}", doc.id);
-            println!("{}", diff);
+        if let fpm::FileFound::FTDDocument(doc) = doc {
+            if doc.id.starts_with(".history") {
```

Comment:
Ideally such files should have been ignored earlier part of the pipeline.

Reply:
Also we should handle binary and text files differently among non ftd documents and show proper diff of text files.